src/codegen/program.lisp

```diff
@@ -189,9 +196,7 @@
                      :for return-type := (typed-node-type (typed-node-abstraction-subexpr node))
                      :append (compile-function name vars type return-type subexpr env))))
 
-          ((and (= 1 (length scc-typed-bindings))
-                t ;; TODO: We need to check that the variable is non-self-recursive using free-variables
```

Comment:
Does this get checked earlier on now?

Reply:
Yeah this is now checked in derive-type

Reply:
See [https://github.com/coalton-lang/coalton/blob/main/src/typechecker/derive-type.lisp#L315-L322](https://github.com/coalton-lang/coalton/blob/main/src/typechecker/derive-type.lisp#L315-L322)